pkg/osquery/tables/apple_silicon_security_policy/parser.go

```diff
@@ -0,0 +1,171 @@
+// OS Pairing Status                             : Not Paired
+// Local Policy Nonce Hash                 (lpnh): A8D3EC575A03E7F58**REDACTED**
+// Remote Policy Nonce Hash                (rpnh): FAD13B6348B0B0FB0**REDACTED**
+// Recovery OS Policy Nonce Hash           (ronh): F7FDBA24525C17FFA**REDACTED**
+//
+// Local policy:
+// Pairing Integrity                             : Valid
+// Signature Type                                : BAA
+// Unique Chip ID                          (ECID): 0x1D68A**REDACTED**
+// Board ID                                (BORD): 0x24
+// Chip ID                                 (CHIP): 0x8103
+// Certificate Epoch                       (CEPO): 0x1
+// Security Domain                         (SDOM): 0x1
+// Production Status                       (CPRO): 1
+// Security Mode                           (CSEC): 1
+// OS Version                              (love): 21.1.559.0.0,0
+// Volume Group UUID                       (vuid): 8462458F-944E-**REDACTED**
+// KEK Group UUID                          (kuid): 41196911-E654-**REDACTED**
+// Local Policy Nonce Hash                 (lpnh): A8D3EC575A03E7F58**REDACTED**
+// Remote Policy Nonce Hash                (rpnh): FAD13B6348B0B0FB0**REDACTED**
+// Next Stage Image4 Hash                  (nsih): 19F3A3DC16816A9FC**REDACTED**
+// User Authorized Kext List Hash          (auxp): absent
+// Auxiliary Kernel Cache Image4 Hash      (auxi): absent
+// Kext Receipt Hash                       (auxr): absent
+// CustomKC or fuOS Image4 Hash            (coih): absent
+// Security Mode:               Full       (smb0): absent
+// 3rd Party Kexts Status:      Disabled   (smb2): absent
+// User-allowed MDM Control:    Disabled   (smb3): absent
+// DEP-allowed MDM Control:     Disabled   (smb4): absent
+// SIP Status:                  Enabled    (sip0): absent
+// Signed System Volume Status: Enabled    (sip1): absent
+// Kernel CTRR Status:          Enabled    (sip2): absent
+// Boot Args Filtering Status:  Enabled    (sip3): absent
+func parseBootPoliciesOutput(reader io.Reader) map[string]interface{} {
+	var volumeGroup string
+	var volumeGroupData []map[string]interface{}
+	results := make(map[string]interface{})
+
+	// This function is called when either a new volume group has been found, or we
+	// have reached the end of the boot policy output. It's purpose is to add an
+	// entry for the most recent volume group processed, if it exists.
+	addVolumeGroupIfNeeded := func() {
+		if volumeGroup != "" && len(volumeGroupData) > 0 {
+			results[volumeGroup] = volumeGroupData
+		}
+		volumeGroupData = nil
+	}
+
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// Always look first for matches with the volume group regexp
+		// Once found, it will be the volume_group column associated with subsequent rows
+		m := volumeGroupRegexp.FindAllStringSubmatch(line, -1)
+		if len(m) == 1 && len(m[0]) == 3 {
```

Comment:
I'm a little confused by this. `FindAllStringSubmatch` returns a slice of matches, but this is scanning a line. Why not use `FindStringSubmatch` ?

Reply:
I guess it's pretty similar, `len(m) == 1` vs `m != nil`

Reply:
I probably copy pasted this from similar table code. Pretty similar but if the intent is more clear one way or the other I'm happy to change.